zhmcclient/_metrics.py

```diff
@@ -906,21 +898,13 @@ def resource(self):
                     Partition, [])
             resource = partitions[0]
         elif resource_class == 'adapter':
-            cpc_list = self.client.cpcs.list()
-            adapter_managers = []
-            for cpc in cpc_list:
-                adapter_managers.append(cpc.adapters)
-                try:
-                    filter_args = {'object-uri': resource_uri}
-                    resource = cpc.adapters.find(**filter_args)
-                    break
-                except NotFound:
-                    pass  # Try next CPC
-            else:
+            adapters = self.client.consoles.console.list_permitted_adapters(
```

Comment:
This operation was introduced with 2.16.0 and in classic mode it does not return the adapters of older systems. Do we need a fallback path here?

Reply:
Very good point. I thought that all our `list_permitted_...()` method have implemented the fallback, but that is not the case. Needs to be done here.

Reply:
Done.